MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -709,6 +722,121 @@ std::string EnggDiffFittingPresenter::functionStrFactory(
   return functionStr;
 }
 
+void EnggDiffFittingPresenter::browsePeaksToFit() {
+  try {
+    QString prevPath = QString::fromStdString(m_view->focusingDir());
+    if (prevPath.isEmpty()) {
+      prevPath = QString::fromStdString(m_view->getPreviousDir());
+    }
+
+    std::string path = m_view->getOpenFile(prevPath.toStdString());
+
+    if (path.empty()) {
+      return;
+    }
+
+    m_view->setPreviousDir(path);
+
+    std::string peaksData = readPeaksFile(path);
+
+    m_view->setPeakList(peaksData);
+  } catch (...) {
```

Comment:
`catch(...)` is a bit too wild, `runtime_error`, `invalid_argument`, etc. are more reasonable.